numpy/core/_add_newdocs.py

```diff
@@ -829,6 +830,16 @@
         Specifies the minimum number of dimensions that the resulting
         array should have.  Ones will be pre-pended to the shape as
         needed to meet this requirement.
+    like : array_like
+        Reference array-like object created by a downstream library, such
+        as CuPy, Dask, xarray, etc. If specified, the function will be
+        dispatched to the downstream library, provided it supports the
+        __array_function__ protocol. This will bypass the NumPy's
+        implementation and be computed by the downstream library, the
+        result is thus returned by that library, which may be of the
+        same type as the array provided here, or any other type.
```

Comment:
These are implementation details. Perhaps replace them with a note linking to the NEP?

Reply:
I thought about that, but the NEP is also too technical for users IMO, that's why I avoided it. What do you think of replacing the whole block by the suggestion below?

```
Reference object created by a library implementing ndarray-like API.
If specified, the function will be dispatched to the library implementing
the reference object, provided it supports the __array_function__ protocol.
```